pkg/managed/clusterroles_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package managed
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-operator/pkg/constants"
+	"github.com/verrazzano/verrazzano-operator/pkg/monitoring"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/types"
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+	rbacv1 "k8s.io/api/rbac/v1"
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+	"testing"
+)
+
+func TestCleanupOrphanedClusterRoles(t *testing.T) {
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster3"]
+
+	cr := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "orphaned",
+			Labels: map[string]string{
+				"verrazzano.binding": "testBinding",
+				"verrazzano.cluster": "cluster3",
+			},
+		},
+	}
+	_, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().Create(context.TODO(), &cr, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("can't create cluster role: %v", err)
+	}
+
+	CleanupOrphanedClusterRoles(modelBindingPair, clusterConnections)
+
+	_, err = clusterConnection.KubeClient.RbacV1().ClusterRoles().Get(context.TODO(), "orphaned", metav1.GetOptions{})
+	if err == nil {
+		t.Fatal("expected cluster role to be cleaned up")
+	}
+}
+
+func TestCreateClusterRoles(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	CreateClusterRoles(modelBindingPair, clusterConnections)
+
+	list, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("can't list cluster roles: %v", err)
+	}
+	assert.Equal(1, len(list.Items))
```

Comment:
i am guessing you are assuming that it is safe to assume that there were no clusterroles because you are using a mock/fake object? is that in fact safe? maybe add a comment to explain why?

Reply:
Yes the test utilities give an assumed starting state.  Yes somebody could change the test utilities but then they will break a bunch of tests and they will either have to adapt the utilities, write new ones or fix the broken tests.  The utilities were something to get up and running quickly with writing tests.  It is assumed that they will improve as needed.